lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -122,6 +142,24 @@ collectAllFormalResultsInTypeOrder(SILFunction &function,
                                                : indResults[indResIdx++]);
 }
 
+/// Given a function, gather all of its direct results in an order defined by
+/// its result type. Note that "formal results" refer to result values in the
+/// body of the function, not at call sites.
+static void
+collectAllDirectResultsInTypeOrder(SILFunction &function,
+                                   SmallVectorImpl<SILValue> &results) {
+  SILFunctionConventions convs(function.getLoweredFunctionType(),
+                               function.getModule());
+  auto *retInst = cast<ReturnInst>(function.findReturnBB()->getTerminator());
+  auto retVal = retInst->getOperand();
+  if (auto *tupleInst =
+      dyn_cast_or_null<TupleInst>(retVal->getDefiningInstruction()))
```

Comment:
Simplify and move to the end of the previous line.
```suggestion
      dyn_cast<TupleInst>(retVal))
```